lib/logstash/filters/date.rb

```diff
@@ -20,256 +25,244 @@
-  # weekday names (pattern with `EEE`).
-  #
-  # If not specified, the platform default will be used but for non-english platform default
-  # an english parser will also be used as a fallback mechanism.
-  config :locale, :validate => :string
-
-  # The date formats allowed are anything allowed by Joda-Time (java time
-  # library). You can see the docs for this format here:
-  #
-  # http://joda-time.sourceforge.net/apidocs/org/joda/time/format/DateTimeFormat.html[joda.time.format.DateTimeFormat]
-  #
-  # An array with field name first, and format patterns following, `[ field,
-  # formats... ]`
-  #
-  # If your time field has multiple possible formats, you can do this:
-  # [source,ruby]
-  #     match => [ "logdate", "MMM dd YYY HH:mm:ss",
-  #               "MMM  d YYY HH:mm:ss", "ISO8601" ]
-  #
-  # The above will match a syslog (rfc3164) or `iso8601` timestamp.
-  #
-  # There are a few special exceptions. The following format literals exist
-  # to help you save time and ensure correctness of date parsing.
-  #
-  # * `ISO8601` - should parse any valid ISO8601 timestamp, such as
-  #   `2011-04-19T03:44:01.103Z`
-  # * `UNIX` - will parse *float or int* value expressing unix time in seconds since epoch like 1326149001.132 as well as 1326149001
-  # * `UNIX_MS` - will parse **int** value expressing unix time in milliseconds since epoch like 1366125117000
-  # * `TAI64N` - will parse tai64n time values
-  #
-  # For example, if you have a field `logdate`, with a value that looks like
-  # `Aug 13 2010 00:03:44`, you would use this configuration:
-  # [source,ruby]
-  #     filter {
-  #       date {
-  #         match => [ "logdate", "MMM dd YYYY HH:mm:ss" ]
-  #       }
-  #     }
-  #
-  # If your field is nested in your structure, you can use the nested
-  # syntax `[foo][bar]` to match its value. For more information, please refer to
-  # <<logstash-config-field-references>>
-  config :match, :validate => :array, :default => []
-
-  # Store the matching timestamp into the given target field.  If not provided,
-  # default to updating the `@timestamp` field of the event.
-  config :target, :validate => :string, :default => "@timestamp"
-
-  # Append values to the `tags` field when there has been no
-  # successful match
-  config :tag_on_failure, :validate => :array, :default => ["_dateparsefailure"]
-
-  # LOGSTASH-34
-  DATEPATTERNS = %w{ y d H m s S }
-
-  public
-  def initialize(config = {})
-    super
-
-    @parsers = Hash.new { |h,k| h[k] = [] }
-  end # def initialize
-
-  public
-  def register
-    require "java"
-    if @match.length < 2
-      raise LogStash::ConfigurationError, I18n.t("logstash.agent.configuration.invalid_plugin_register",
-        :plugin => "filter", :type => "date",
-        :error => "The match setting should contains first a field name and at least one date format, current value is #{@match}")
-    end
+module LogStash
```

Comment:
I think Jordan preferred the `module LogStash module Filters class Date < LogStash::Filters::Base` style declaration because it makes the whole file less indented.